### DISCUSSION TITLE
Tile Loading Odd Behavior

### DISCUSSION THREAD

**Discussion by bighamj** [iTwin/itwinjs-core#4921] [Q&A]
Using the RealityData sample (https://www.itwinjs.org/sandboxes/iTwinPlatform/Reality%20Data), I was able to use some tiles from data I've uploaded to the iTwin ecosystem. However, it seems to load all points at once and never re-render/load the tiles. For example, I have points that are larger and smaller pixels size by side: 

![image](https://user-images.githubusercontent.com/66083892/212171483-73e0cd62-1def-4317-a7cc-363a77f33d17.png)

I was chasing changing the point size based on a slider like the example, but realize that my data itself is being displayed oddly.

**Comment by pmconne** [comment]
Hi @bighamj. That certainly doesn't look right, though I don't know what this point cloud is supposed to look like.
Does it look correct in other viewers? If so, which ones? Do you have any screenshots for comparison?
How did you produce the tiles? Are the tiles in .pnts format, or some other format? If they are .pnts, would it be possible to supply the root JSON file describing the tile hierarchy?

**Reply by bighamj** [reply]
Hello there! 

I use Cesium Ion to produce the tileset from an LAS file. See below from their test viewer: 

![image](https://user-images.githubusercontent.com/66083892/212996681-3b9b9828-22e3-4488-bf64-a3e930e4e718.png)

Files are pnts. I am using their new export, which I've found zip the files when transferring to S3, but I've unzipped them before uploading to the Blob.

JSON here:

[tileset.json.zip](https://github.com/iTwin/itwinjs-core/files/10438841/tileset.json.zip)

**Reply by pmconne** [reply]
Thanks @bighamj. I notice the tileset is using additive refinement, which probably explains why you see all the points simultaneously. 
`    "refine": "ADD"`
If you zoom out far enough, do some of the points disappear?

iTwin.js by default computes a size for each point based on the density of the points in the tile containing it. For additive refinement, that probably isn't useful, since the point density in that case is determined by multiple overlapping tiles.
We recently (in iTwin.js 3.5) [added APIs](https://www.itwinjs.org/changehistory/) that allow you to customize how your point cloud looks. You might try using a [fixed pixel size](https://www.itwinjs.org/reference/core-common/displaystyles/pointcloudsizemode/) like Cesium is using in your screenshot.

If you're comfortable sending the point cloud confidentially to paul `dot` connelly `at` bentley `dot` com, I would be happy to experiment and make sure we're handling the additive refinement correctly.
Otherwise please let us know if adjusting the point size mode resolves the problem.

**Reply by bighamj** [reply]
Email sent. Thanks for your help!

**Comment by pmconne** [comment]
Here's [a sandbox](https://www.itwinjs.org/sandboxes/pmconne/point-cloud-customization) demonstrating how to change some of the point cloud display settings, e.g., making each point draw at a fixed size in pixels.
